docs/move/book/loops.md

```diff
@@ -114,6 +114,45 @@ fun pick(
 }
 ```
 
+## The `for` expression
+
+The `for` expression iterates over a range defined using integer-typed `lower_bound` (inclusive) and `upper_bound` (non-inclusive) expressions, executing its loop body for each element of the range. `for` is designed for scenarios where the number of iterations of a loop is determined by a specific range.
+
+Here is an example of a `for` loop that computes the sum of the elements in a range from `0` to `n-1`:
+
+```move
+fun sum(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..n) {
+        sum = sum + i;
+    };
+
+    sum
+}
+```
+
+### `break` and `continue` in `for` loops
+
+Similar to `while` loops, the `break` expression can be used in `for` loops to exit prematurely. The `continue` expression can be used to skip the current iteration and move to the next. Here's an example that demonstrates the use of both `break` and `continue`. The loop will iterate through numbers from `0` to `n`, summing up them up. It will skip numbers that are divisible by `3` (using `continue`) and stop when it encounters a number greater than `10` (using `break`):
+
+```move
+fun sum_conditional(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..(n + 1)) {
```

Comment:
Use a different variable than `i` here, to show noobs that the index variable can have any name.  Perhaps `count` or `idx`.

Reply:
I used `iter`